### ISSUE TITLE
Should train_chain_data_cache_path be a required argument?

### ISSUE THREAD

**Issue by jonathanking** [aqlaboratory/openfold#251]
Although the current argparse parser allows the user to not pass a value for `train_chain_data_cache_path`, the current implementation of `data_modules.OpenFoldDataset` (specifically the inner function, `looped_samples`) assumes that the cache object is not None. If the user does not supply a cache path, then the training script simply fails with a StopIteration, as it tries to get a cache entry from a None object on line 371:

https://github.com/aqlaboratory/openfold/blob/59277de16825cfdafe37033012d0530595b9ad6d/openfold/data/data_modules.py#L360-L374

It seems like OpenFold's datasets have been built to support parsing structure files on the fly as well, so which of the two options would be preferred going forward? 1) make train_chain_data_cache_path required, so the user does not have an unexpected failure when the data is loaded, or 2) Adding support in `OpenFoldDataset/looped_samples` for the case that the cache is None?

Happy to help implement something either way!

**Comment by gahdritz** [comment]
Good point. I think the former option is better, since performance suffers greatly if you have to parse mmCIF files all the time.

**Comment by jonathanking** [comment]
Hey @gahdritz, thanks for following up! 

I'm a little confused. When training, are training set structures supposed to be parsed during every training step, regardless of any precomputed caches?

To me, it seems like the training loop is not using any sort of indexing/caching for protein structure/coordinate data, which doesn't seem like the intended behavior.

Let me explain why I think so: 

1. In order to avoid parsing a file in OpenFoldSingleDataset, you must pass in a _structure_index object. When setting up the dataset [here](https://github.com/aqlaboratory/openfold/blob/42e71db7fa327e0810eb0e371abc9f82aa9b7a6a/openfold/data/data_modules.py#L620), no structure index is provided for the training dataset. 
    - Even if the structure index is passed to the training dataset, the _structure_index entry simply points to the file path, and the string must be read and parsed.
3. I also considered the following. The `chain_data_cache.json` object created by `generate_chain_data_cache.py` is a cache storing sequences, resolutions, and cluster sizes for each protein.
    - This does not contain ground truth atomic coordinates, so this data structure must not serve the purpose of caching all data necessary to train the model.


Thanks so much for helping me understand the code better!
